dlt/pipeline/pipeline.py

```diff
@@ -872,11 +881,7 @@ def _extract_source(self, storage: ExtractorStorage, source: DltSource, max_para
         self._schema_storage.save_import_schema_if_not_exists(source_schema)
 
         # get the current schema and merge tables from source_schema
-        # note we are not merging props like max nesting or column propagation
-        for table in source_schema.data_tables(include_incomplete=True):
-            pipeline_schema.update_schema(
-                pipeline_schema.normalize_table_identifiers(table)
-            )
+        pipeline_schema.update_schema(source_schema)
```

Comment:
on one hand - this is good idea. on the other hand the implementation must be similar to those in normalize py
- create partial (diff table)
- apply only if there are any changes
- apply the schema contract to check if allowed, see my big comment at the end